src/tests/integration/test-data/cse-lemmas.k

```diff
@@ -55,6 +55,12 @@ module CSE-LEMMAS
        andBool X +Int 1 <=Int A
        [concrete(X), simplification, preserves-definedness]
 
+    rule { #asWord( B1 +Bytes B2 ) #Equals #asWord( B3 +Bytes B4 ) } 
+      => { B1 #Equals B3 } #And { B2 #Equals B4 }
+    requires lengthBytes(B1 +Bytes B2) <=Int 32
+    andBool lengthBytes(B1) ==Int lengthBytes(B3) 
+    andBool lengthBytes(B2) ==Int lengthBytes(B4) [simplification]
```

Comment:
You only need one of these two, I think.

Reply:
Actually, note [this KEVM simplification](https://github.com/runtimeverification/evm-semantics/blob/bd4e443f80d827355236fb2cd4d3a4aaec75d075/kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/bytes-simplification.k#L21). I think that it would be better not to repeat its logic, but to instead use
```
rule { #asWord ( X ) #Equals #asWord ( Y ) } => #Top
  requires X ==K Y
  [simplification]
```
which would then piggyback onto the existing ones for `Bytes`.

Reply:
Replaced with the new lemma.

Reply:
The combination of the two lemmas
```
    rule { #asWord ( X ) #Equals #asWord ( Y ) } => #Top
      requires X ==K Y
      [simplification]
```
and
```
    rule [bytes-equal-concat-split-ml]:
      A:Bytes +Bytes B:Bytes ==K C:Bytes +Bytes D:Bytes => (A ==K C) andBool (B ==K D)
      requires lengthBytes(A) ==Int lengthBytes(C)
        orBool lengthBytes(B) ==Int lengthBytes(D)
      [simplification]
```
Does not eliminate the branching we were trying to eliminate because what my original lemma was doing was getting byte equality out of word equality.
Would this one be acceptable instead to work in tandem with `bytes-equal-concat-split-ml`?
```
    rule { #asWord ( X ) #Equals #asWord ( Y ) } => { X #Equals Y }
      requires lengthBytes(X) ==Int lengthBytes(Y) andBool lengthBytes(X) <=Int 32
      [simplification]
```

Reply:
Oh that's interesting. Yes, that should work.